run-container.sh

```diff
@@ -1,42 +1,151 @@
+	'OUTPUT_LOC is required. Use the -o option to specify the output location for the test results.'
+)
 
-usage() {
-	echo "$0 -k KUBECONFIG -t TNFCONFIG -o OUTPUT_LOC SUITE [... SUITE]"
-	echo "Configure and run the containerised TNF test offering."
-	echo "  e.g."
-	echo "    $0 -k ~/.kube/config -t ~/tnf/config -o ~/tnf/output diagnostic generic"
-	echo "  will run the diagnostic and generic tests, and output the result into"
-	echo "  ~/tnf/output on the host."
-	echo ""
-	echo "Allowed tests are listed in the README."
-	echo "Note: Tests muse be specified after all other arguments"
+CONTAINER_TNF_DIR=/usr/tnf
+CONTAINER_TNF_KUBECONFIG_FILE_BASE_PATH="$CONTAINER_TNF_DIR/kubeconfig/config"
+
+usage () {
+	read -d '' usage_prompt <<- EOF
+	Usage: $0 -t TNFCONFIG -o OUTPUT_LOC [ -k KUBECONFIG ] SUITE [ ... SUITE ]
+
+	Configure and run the containerised TNF test offering.
+
+	Options
+	  -t: set the directory containing TNF config files set up for the test.
+	  -o: set the output location for the test results.
+	  -k: set path to one or more local kubeconfigs, separated by a colon.
+	      The -k option takes precedence, overwriting the results of local kubeconfig autodiscovery.
+	      See the 'Kubeconfig lookup order' section below for more details.
+
+	Kubeconfig lookup order
+	  1. If -k is specified, use the paths provided with the -k option.
+	  2. If -k is not specified, use paths defined in \$KUBECONFIG on the underlying host.
+	  3. If no paths are defined, use the default kubeconfig file located in '\$HOME/.kube/config'
+	     (currently: $HOME/.kube/config).
+
+	Examples
+	  $0 -t ~/tnf/config -o ~/tnf/output diagnostic generic
+
+	  Because -k is omitted, $(basename $0) will first try to autodiscover local kubeconfig files.
+	  If it succeeds, the diagnostic and generic tests will be run using the autodiscovered configuration.
+	  The test results will be saved to the '~/tnf/output' directory on the host.
+
+	  $0 -k ~/.kube/ABC:~/.kube/DEF -t ~/tnf/config -o ~/tnf/output diagnostic generic
+
+	  The command will bind two kubeconfig files (~/.kube/ABC and ~/.kube/DEF) to the TNF container,
+	  run the diagnostic and generic tests, and save the test results into the '~/tnf/output' directory
+	  on the host.
 
+	Test suites
+	  Allowed tests are listed in the README.
+	  Note: Tests must be specified after all other arguments!
+	EOF
+
+	echo -e "$usage_prompt"
 }
 
 usage_error() {
 	usage
 	exit 1
 }
 
-CONTAINER_TNF_DIR=/usr/tnf
+check_cli_required_num_of_args() {
+	if (($# < 5)); then
+		usage_error
+	fi;
+}
+
+check_required_vars() {
+	local var_missing=false
+
+	for index in "${!REQUIRED_VARS[@]}"; do
+		var=${REQUIRED_VARS[$index]}
+		if [[ -z ${!var} ]]; then
+			error_message=${REQUIRED_VARS_ERROR_MESSAGES[$index]}
+			echo "$0: error: $error_message" 1>&2
+			var_missing=true
+		fi
+	done
+
+	if $var_missing; then
+		echo ""
+		usage_error
+	fi
+}
```

Comment:
Instead of differentianting between _required_ and _optional_ CLI parameters, I chose to check for the existence of variables required to run the TNF container.
This seems to me to be a much simpler approach, especially when dealing with autodiscovery for some of our variables.

I've left in the naive checking for the required number of arguments.